src/inverted_index.c

```diff
@@ -561,16 +561,11 @@ SKIPPER(seekFreqOffsetsFlags) {
       qint_decode4(br, &did, &freq, (uint32_t *)&fm, &offsz);
       Buffer_Skip(br, offsz);
       lastId = (did += lastId);
-      if (did > expid) {
+      if (!(ctx->num & fm)) {
```

Comment:
Don't dereference ctx; first check did >= expid as before

Reply:
This was actually the issue, you did not check that the doc not match the field mask

Reply:
hrm, I see. Can you cache the variable `ctx->num` as `userFm` locally, and use that? I'd also like to see some comments regarding why skipper is called in a loop (it took me a while to figure out why...)

Reply:
I think the compiler probably optimize it anyway, but yeah I will change it...

Reply:
By the way a much better optimization would be to use a binary search on the blocks.

Reply:
We already do a binary search on the block to find which block contains the ID, but within the individual block we must read sequentially.

Reply:
From looking at the code, not binary search is done ..

Reply:
Sorry, found it :+1: